platform/broadcom/sonic-platform-modules-dell/s6100/scripts/s6100_ssd_upgrade_status.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash
+
+SSD_FW_UPGRADE="/host/ssd_fw_upgrade"
+
+if [ -e $SSD_FW_UPGRADE/GPIO7_high ]; then
+    systemctl start --no-block s6100-ssd-monitor.timer
+    exit 0
+fi
+
+if [ -e $SSD_FW_UPGRADE/GPIO7_low ] || [ -e $SSD_FW_UPGRADE/GPIO7_error ]; then
+    exit 0
+fi
+
+[ ! -d $SSD_FW_UPGRADE ] && mkdir $SSD_FW_UPGRADE
+
+SSD_UPGRADE_LOG="$SSD_FW_UPGRADE/upgrade.log"
+
+SMART_CMD=`smartctl -a /dev/sda`
+
+SSD_FW_VERSION=$(echo "$SMART_CMD" | grep "Firmware Version" | awk '{print $NF}')
+SSD_MODEL=$(echo "$SMART_CMD" | grep "Device Model" | awk '{print $NF}')
+
+if [ -e $SSD_FW_UPGRADE/GPIO7_pending_upgrade ]; then
+    if [ $SSD_FW_VERSION == "S141002C" ] || [ $SSD_FW_VERSION == "S16425c1" ]; then
+        # If SSD Firmware is not upgraded
+        exit 0
+    fi
+fi
+
+echo "$0 `date` SSD FW upgrade logs post reboot." >> $SSD_UPGRADE_LOG
+
+iSMART="/usr/local/bin/iSMART_64"
+iSMART_OPTIONS="-d /dev/sda"
+iSMART_CMD=`$iSMART $iSMART_OPTIONS`
+
+SSD_UPGRADE_STATUS1=`io_rd_wr.py --set --val 06 --offset 210; io_rd_wr.py --set --val 09 --offset 211; io_rd_wr.py --get --offset 212`
+SSD_UPGRADE_STATUS1=$(echo "$SSD_UPGRADE_STATUS1" | awk '{print $NF}')
+
+SSD_UPGRADE_STATUS2=`io_rd_wr.py --set --val 06 --offset 210; io_rd_wr.py --set --val 0A --offset 211; io_rd_wr.py --get --offset 212`
+SSD_UPGRADE_STATUS2=$(echo "$SSD_UPGRADE_STATUS2" | awk '{print $NF}')
+
+if [ $SSD_UPGRADE_STATUS1 == "2" ]; then
+    rm -rf $SSD_FW_UPGRADE/GPIO7_*
+    touch $SSD_FW_UPGRADE/GPIO7_error
+
+    echo "$0 `date` Upgraded to unknown version after first mp_64 upgrade." >> $SSD_UPGRADE_LOG
+
+elif [ $SSD_UPGRADE_STATUS2 == "2" ];then
+    rm -rf $SSD_FW_UPGRADE/GPIO7_*
+    touch $SSD_FW_UPGRADE/GPIO7_error
+
+    echo "$0 `date` Upgraded to unknown version after second mp_64 upgrade." >> $SSD_UPGRADE_LOG
+
+elif [ $SSD_FW_VERSION == "S141002G" ] || [ $SSD_FW_VERSION == "S16425cG" ]; then
+    # If SSD Firmware is upgraded
+    GPIO_STATUS=$(echo "$iSMART_CMD" | grep GPIO | awk '{print $NF}')
+
+    if [ $GPIO_STATUS != "0x01" ];then
+        logger -p user.crit -t DELL_S6100_SSD_MON "The SSD on this unit is faulty and does not support reboot."
+        logger -p user.crit -t DELL_S6100_SSD_MON "If a reboot is required, please perform a soft-/fast-/warm-reboot."
+        rm -rf $SSD_FW_UPGRADE/GPIO7_*
+        touch $SSD_FW_UPGRADE/GPIO7_low
+        echo "$0 `date` The SSD on this unit is faulty and does not support cold reboot." >> $SSD_UPGRADE_LOG
+        echo "$0 `date` If a reboot is required, please perform a soft-/fast-/warm-reboot." >> $SSD_UPGRADE_LOG
+
+    else
+        rm -rf $SSD_FW_UPGRADE/GPIO7_*
+        touch $SSD_FW_UPGRADE/GPIO7_high
+    fi
+
+    systemctl start --no-block s6100-ssd-monitor.timer
```

Comment:
why do you need start the timer here again?

Reply:
As discussed, this timer needs to be started only on two cases - when SSD FW is being upgraded first time and then on reboots when GPIO7 status is still high.